src/app/chip-zcl-zpro/command-encoder/decoder.c

```diff
@@ -105,14 +105,14 @@ void printApsFrame(EmberApsFrame * frame)
 
 uint16_t extractMessage(uint8_t * buffer, uint16_t buffer_length, uint8_t ** msg)
 {
-    // buffer is at least 16 bytes long and the msg is 3 bytes long and 13 bytes offset into the buffer.
-    // This is hard coded now till we refactor encoder.c to not have this be hardcoded.
+    // The size of an encoded EmberApsFrame is currently 13 bytes.  The message
+    // starts after that.
     uint16_t result = 0;
-    if (msg && buffer_length >= 16)
+    if (msg && buffer_length > 13)
```

Comment:
we should add a constant like `constexpr size_t kEmberAppsFrameSize = 13;` or similar

Reply:
Going to do that, but also filed https://github.com/project-chip/connectedhomeip/issues/1980 to make all this go away.